### ISSUE TITLE
 Constraint constraints/compute.vmExternalIpAccess violated for project

### ISSUE THREAD

**Issue by jal06** [kubeflow/gcp-blueprints#121]
I'd like to deploy a Kubeflow instance version 1.1 following instructions[ here](https://www.kubeflow.org/docs/gke/deploy/management-setup/) 

When applying the manifests for creation of the management cluster, I get an error :

`make apply`

```
# Delete the directory so any resources that have been removed
# from the manifests will be pruned
rm -rf .build
mkdir -p .build/
mkdir -p .build/cluster
kustomize build ./instance/cluster -o .build/cluster
anthoscli apply --project=HOST_PROJECT -f .build/cluster
Flag --project has been deprecated, GCE project flag is unused and deprecated
W0812 11:37:33.878350    2382 warning.go:14] manifest is using some deprecated features; please run vet for details
I0812 11:37:33.878481    2382 main.go:230] reconcile serviceusage.cnrm.cloud.google.com/Service container.googleapis.com
Unexpected error: error reconciling objects: error reconciling Service:HOST_PROJECT/container.googleapis.com: error fetching service "projects/HOST_PROJECT/services/container.googleapis.com": googleapi: Error 400: <eye3 title='/ResourceCache.GetProjectInfo, INVALID_ARGUMENT'/> APPLICATION_ERROR;google.internal.cloud.resourcemanager/ResourceCache.GetProjectInfo;Invalid project resource name projects/HOST_PROJECT
com.google.apps.framework.request.StatusException: <eye3 title='INVALID_ARGUMENT'/> generic::INVALID_ARGUMENT: Invalid project resource name projects/HOST_PROJECT;AppErrorCode=3;StartTimeMs=1597225055541;unknown;Deadline(sec)=3.0;ResFormat=AUTOMATIC;ServerTimeSec=0.009967;LogBytes=256;FailFast;EffSecLevel=none;ReqFormat=AUTOMATIC;ReqID=a6bbc7c17ed26696;GlobalID=0;Server=[2002:ae1:e816::]:4442
Makefile:49: recipe for target 'apply' failed
make: *** [apply] Error 1
```

I don't know if it can help, but below is the anthoscli version used 

`anthoscli version`
`Version: version.Version{AnthosCLIVersion:"beta", GitCommit:"9996ec1cd6729a40406a7a300db9ff4b6b5cdc9a", GitTag:"0.2.4", BuildDate:"2020-07-16T19:27:40Z", GoOs:"linux", GoArch:"amd64"}`

Did I miss something ?

**Comment by jal06** [comment]
The solution is described in this issue [2123](https://github.com/kubeflow/website/issues/2123)
I run make set-values before running make apply and it works 



However I had to modify Makefile for removing the 1rst line ( set-values: ) in the set-values def.

At the end, there is also a warning :
```
...
W0812 17:31:17.516825   11429 nodepool.go:441] unsupported change: {Path:Config.WorkloadMetadataConfig.Mode Actual:GKE_METADATA Expected:}
Reconciliation completed successfully

```
I don't know what means this warning and if it is important

**Comment by jal06** [comment]
The cluster is not started. I see this error in the console  
`Not all instances running in IGM after 30.687239058s. Expect 1. Current errors: [CONDITION_NOT_MET]: Instance 'gke-xxxxxx-xxxxxx-pool-aaaaaaaa-bbbb' creation failed: Constraint constraints/compute.vmExternalIpAccess violated for project 123456789012. Add instance projects/xxxxxxx/zones/europe-west1-b/instances/gke-xxxxxx-xxxxxx-pool-aaaaaaaa-bbbb to the constraint to use external IP with it. `

**Comment by jal06** [comment]
removing the 1rst line ( set-values: ) in the set-values def is solved by PR [120](https://github.com/kubeflow/gcp-blueprints/pull/120)

But cluster doesn't start due to error above

**Comment by Bobgy** [comment]
@jal06 looks like your org has a policy to not allow cluster with external ip

Would you mind creating Kustomize overlays in the `instance` folder for the cluster instance, so that it works for your org?
The resource specs are consumed by https://cloud.google.com/config-connector/docs/overview service.
You may refer to cluster spec reference in https://cloud.google.com/config-connector/docs/reference/resource-docs/container/containercluster.

**Comment by jal06** [comment]
@Bobgy thanks for your answer.

I will ask my admin team to have more infos about our policy.

I guess I must enable the ContainerCluster.networkPolicy and set a value in ContainerCluster.networkPolicy.provider . 
Am I right ?
 
```
networkPolicy :
  enabled: boolean
  provider: string
```

Please, could you elaborate why we need to allow external IP here ?

**Comment by Bobgy** [comment]
@jal06 it's not required, you can follow guide for private cluster in https://github.com/kubeflow/gcp-blueprints/blob/master/kubeflow/deploy_private.md.

**Comment by jal06** [comment]
@Bobgy thanks for your help. My admin team helped me to solve that issue.
So I tried to install again 1.1 and it works perfectly


About the warning mentionned above `W0812 17:31:17.516825   11429 nodepool.go:441] unsupported change: {Path:Config.WorkloadMetadataConfig.Mode Actual:GKE_METADATA Expected:}` , I think the root cause is that I didn't set the right values for the variable NAME.
My concern is that, when executing this step, I know that I have to create 2 K8S clusters :
- the management cluster
- the Kubeflow cluster
and I must set twice a value for the variable NAME. 

So instead of setting the name of the management cluster to both `kpt cfg set ./instance name NAME` and `kpt cfg set ./upstream/management name NAME` ,  I set the name of the kubeflow cluster to `kpt cfg set ./instance name NAME` and the name of the management cluster to` kpt cfg set ./upstream/management name NAME`

Maybe it could make sense if 
- instead of NAME, the name of the variable is MGMNT_NAME (or CNRM_NAME ) ?
- we set this variable only once , such as kpt cfg set ./ name CNRM_NAME

For a confirmed admin, it looks obvious that NAME is the name of the management cluster. 
IMHO, for a newbie, it doesn't look obvious.
